packages/storybook/stories/va-search-input.stories.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { VaSearchInput } from '@department-of-veterans-affairs/web-components/react-bindings';
+import { getWebComponentDocs, propStructure } from './wc-helpers';
+import { generateEventsDescription } from './events';
+
+VaSearchInput.displayName = 'VaSearchInput';
+
+const searchDocs = getWebComponentDocs('va-search-input');
+
+export default {
+  title: 'Components/va-search-input',
+  parameters: {
+    componentSubtitle: 'Search Input web component',
+    docs: {
+      description: {
+        component: generateEventsDescription(searchDocs),
+      },
+    },
+  },
+};
+
+const Template = ({ 'button-text': buttonText, value, label, suggestions }) => (
+  <VaSearchInput
+    buttonText={buttonText}
+    value={value}
+    label={label}
+    onInput={e => console.log(e)}
+    onSubmit={e => console.log(e)}
+    suggestions={suggestions}
+  />
+);
+
+const defaultArgs = {
+  'button-text': undefined,
+  'value': 'benefits',
+  'label': 'Search',
+  'suggestions': undefined,
+};
+
+export const Default = Template.bind({});
+Default.args = {
+  ...defaultArgs,
+};
+Default.argTypes = propStructure(searchDocs);
+
+export const WithButtonText = Template.bind({});
+WithButtonText.args = {
+  ...defaultArgs,
+  'button-text': 'Search',
+};
+
+const TypeaheadTemplate = ({ value, suggestions }) => {
+  const [text, setText] = useState(value);
+  const [latestSuggestions, setLatestSuggestions] = useState(suggestions);
+
+  const handleInput = e => {
+    console.log(e);
+    // event.composedPath()[0].value outside of React
+    setText(e.nativeEvent.composedPath()[0].value);
+  };
+
+  const handleSubmit = e => {
+    // e.detail.value outside of React
+    console.log(e.nativeEvent.detail.value);
+  };
+
+  /**
+   * Mock suggestions
+   * Provides suggestions for the following values: for, form, forms
+   * All other values will return an empty array
+   */
+  useEffect(() => {
+    switch (text) {
+      case 'for':
+        setTimeout(() => {
+          setLatestSuggestions([
+            'form',
+            'form finder',
+            'form search',
+            'foreign study',
+            'forever gi bill',
+          ]);
+        }, 1000);
+        break;
+      case 'form':
+        setTimeout(() => {
+          setLatestSuggestions(['form', 'forms', 'form finder', 'form search']);
+        }, 1000);
+        break;
+      case 'forms':
+        setTimeout(() => {
+          setLatestSuggestions(['forms']);
+        }, 1000);
+        break;
+      default:
+        setTimeout(() => {
+          setLatestSuggestions([]);
+        }, 1000);
+        break;
+    }
```

Comment:
I know this is just a storybook thing, but we may be able to get rid of the case statement and make this easier to maintain with something like:

```js
const suggestions = [
  'form', 
  'form finder', 
  'form search', 
  'foreign study', 
  'forever gi bill'
];

setLatestSuggestions(suggestion.filter(s => s.includes(text)));
```

Reply:
This request has been addressed [here](https://github.com/department-of-veterans-affairs/component-library/pull/391).